### ISSUE TITLE
Floating point exception (core dumped)

### ISSUE THREAD

**Issue by Kitsunetic** [traveller59/spconv#725]
I always get floating point exception while I'm using SubMConv3d.

Here is my test code:
```py
import torch as th
from spconv.pytorch import SubMConv3d, SparseConvTensor

xyz = th.randint(0, 32, (1000, 4), dtype=th.int64, device='cuda')
xyz[:, 0] = 0
feat = th.randn(1000, 32, device='cuda', dtype=th.float32)
sp = SparseConvTensor(feat, xyz, (32, 32, 32), 1, 1, 1)

conv = SubMConv3d(32, 64, 3).cuda()
conv(sp)

>>> Floating point exception (core dumped)
```

I'm using PyTorch 2.3.0 with CUDA 11.8, and spconv-cu18==2.3.6.
Is there something wrong in my code, or someone knows the clue?

I have tested with A5000 and RTX 2080Ti GPUs but the result was always same.

**Comment by shim94kr** [comment]
I'm experiencing the exact same issue. 

I've found that it works fine with kernel_size=1, but consistently crashes with kernel_size=3 or any other size.

**Comment by Kitsunetic** [comment]
> I'm experiencing the exact same issue.
> 
> I've found that it works fine with kernel_size=1, but consistently crashes with kernel_size=3 or any other size.
> 
> @Kitsunetic Have you fixed this issue?

No, I'm still figuring out the solution.

**Comment by shim94kr** [comment]
I found that downgrading PyTorch to version 2.2 resolves the issue.

**Comment by Kitsunetic** [comment]
which cuda version did you use?

**Comment by shim94kr** [comment]
I use CUDA 12.1, and I installed spconv-cu120.

**Comment by Kitsunetic** [comment]
Unfortunately, I'm still getting same issue with my retrial on nvidia/cuda:12.1.0-cudnn8-devel-ubuntu22.04 docker image with Pytorch 2.2.2 with CUDA 12.1. I have tested with both ubuntu 22.04 and 20.04. Couly you give me more detail about your environment?

**Comment by Kitsunetic** [comment]
Thank you for sharing.
However... I'm still getting same error even with environment based on provided yaml file.
I expect this is not only the problem of dependencies, but also entire environment like OS can be related. So, I'm still figuring out the reason.
Anyway, thank you again for your sharing! If you found another clue, please share with me!

**Comment by JunseoMin** [comment]
Hi, 

I have the same issue...
Please share it with me if you solve this issue.

Thanks!

**Comment by Ecalpal** [comment]
same as you
python 3.11, pytorch 2.5.0, cuda 12.1


> I found that downgrading PyTorch to version 2.2.2 resolves the issue.

and this works

**Comment by GCChen97** [comment]
It seems to do with numpy version >= 2.0.0. I installed numpy 1.26.4 and `spconv.pytorch.ops.implicit_gemm` can be called without raising `Floating point exception (core dumped)`. Something is wrong with the arg `masks` of implicit_gemm which is also a numpy array.

**Comment by JunseoMin** [comment]
> It seems to do with numpy version >= 2.0.0. I installed numpy 1.26.4 and `spconv.pytorch.ops.implicit_gemm` can be called without raising `Floating point exception (core dumped)`. Something is wrong with the arg `masks` of implicit_gemm which is also a numpy array.

This worked for me! Thanks!

**Comment by Kitsunetic** [comment]
Thanks @shim94kr @JunseoMin @GCChen97 @Ecalpal !
I solved too.
I will close this issue then.

**Comment by Chinese-Coding** [comment]
> It seems to do with numpy version >= 2.0.0. I installed numpy 1.26.4 and `spconv.pytorch.ops.implicit_gemm` can be called without raising `Floating point exception (core dumped)`. Something is wrong with the arg `masks` of implicit_gemm which is also a numpy array.

Is there any plan to support NumPy >= 2.0.0? This package was proposed almost a year ago, and many foundational packages, such as PyTorch, already support it.

**Comment by abhijeetWork** [comment]
I believe the issue is with numpy>2. My issue got resolved as soon as I forced the installation for numpy<2 before installing spconv.
